rust/lance-encoding/benches/decoder.rs

```diff
@@ -65,33 +55,37 @@ const ENCODING_OPTIONS: EncodingOptions = EncodingOptions {
 fn bench_decode(c: &mut Criterion) {
     let rt = tokio::runtime::Runtime::new().unwrap();
     let mut group = c.benchmark_group("decode_primitive");
+    const NUM_BYTES: u64 = 1024 * 1024 * 128;
+    group.throughput(criterion::Throughput::Bytes(NUM_BYTES));
     for data_type in PRIMITIVE_TYPES {
-        let data = lance_datagen::gen()
-            .anon_col(lance_datagen::array::rand_type(data_type))
-            .into_batch_rows(lance_datagen::RowCount::from(1024 * 1024 * 1024))
-            .unwrap();
-        let lance_schema =
-            Arc::new(lance_core::datatypes::Schema::try_from(data.schema().as_ref()).unwrap());
-        let input_bytes = data.get_array_memory_size();
-        group.throughput(criterion::Throughput::Bytes(input_bytes as u64));
-        let encoding_strategy = default_encoding_strategy(LanceFileVersion::default());
-        let encoded = rt
-            .block_on(encode_batch(
-                &data,
-                lance_schema,
-                encoding_strategy.as_ref(),
-                &ENCODING_OPTIONS,
-            ))
-            .unwrap();
         let func_name = format!("{:?}", data_type).to_lowercase();
+        let num_rows = NUM_BYTES / data_type.primitive_width().unwrap() as u64;
         group.bench_function(func_name, |b| {
+            let data = lance_datagen::gen()
+                .anon_col(lance_datagen::array::rand_type(data_type))
+                .into_batch_rows(lance_datagen::RowCount::from(num_rows))
+                .unwrap();
+            let lance_schema =
+                Arc::new(lance_core::datatypes::Schema::try_from(data.schema().as_ref()).unwrap());
+            let encoding_strategy = default_encoding_strategy(LanceFileVersion::default());
```

Comment:
is this intentional that we only run `bench_decode` against 2.0(LanceFileVersion::default)?

Reply:
For now, yes.  I think I updated one or two benchmarks to run against 2.1 but wasn't trying to update the whole suite yet.